go/plugins/dotprompt/genkit.go

```diff
@@ -216,10 +236,76 @@ func (p *Prompt) Generate(ctx context.Context, pr *PromptRequest, cb func(contex
 		}
 	}
 
-	resp, err := model.Generate(ctx, genReq, cb)
+	resp, err := model.Generate(ctx, mr, pr.Stream)
 	if err != nil {
 		return nil, err
 	}
 
 	return resp, nil
 }
+
+// WithVariables adds variables to pass to the model.
+func WithVariables(variables any) GenerateOption {
+	return func(p *PromptRequest) error {
+		if p.Variables != nil {
+			return errors.New("dotprompt.WithVariables: cannot set variables more than once")
+		}
+		p.Variables = variables
+		return nil
+	}
+}
+
+// WithConfig adds model configuration. If nil will be taken from the prompt config.
+func WithConfig(config *ai.GenerationCommonConfig) GenerateOption {
+	return func(p *PromptRequest) error {
+		if p.Config != nil {
+			return errors.New("dotprompt.WithConfig: cannot set config more than once")
+		}
+		p.Config = config
+		return nil
+	}
+}
+
+// WithContext add context to pass to model, if any.
+func WithContext(context []any) GenerateOption {
+	return func(p *PromptRequest) error {
+		if p.Context != nil {
+			return errors.New("dotprompt.WithContext: cannot set context more than once")
+		}
+		p.Context = context
+		return nil
+	}
+}
+
+// WithModel adds the Model to use. This overrides any model specified by the prompt.
+func WithModel(model ai.Model) GenerateOption {
+	return func(p *PromptRequest) error {
+		if p.ModelName != "" || p.Model != nil {
+			return errors.New("dotprompt.WithModel: config must specify exactly once, either ModelName or Model")
+		}
+		p.Model = &model
+		return nil
+	}
+}
+
+// WithModelName adds the name of the Model to use. This overrides any model specified by the prompt.
+func WithModelName(model string) GenerateOption {
+	return func(p *PromptRequest) error {
+		if p.ModelName != "" || p.Model != nil {
+			return errors.New("dotprompt.WithModelName: config must specify exactly once, either ModelName or Model")
```

Comment:
s/config/model? Here and above. "model must be specified exactly once, either ModelName or Model"